test/test_optim.py

```diff
@@ -56,6 +56,78 @@ def test_errors(self, device, dtype, optim_info):
                 raise NotImplementedError(f"Unknown error type {error_input.error_on}")
 
 
+    @parametrize("on", ["contiguous_params", "noncontiguous_params"])
```

Comment:
nit: "on" is not a great variable name. contiguous_param. [True, False] would be easier to read.